src/gsi/cplr_get_wrf_mass_ensperts.f90

```diff
@@ -1705,6 +1740,338 @@ subroutine general_read_wrf_mass2(this,filename,g_ps,g_u,g_v,g_tv,g_rh,g_cwmr,g_
   return
   end subroutine general_read_wrf_mass2
 
+  subroutine general_read_wrf_mass_dirZDA (this,filename,g_qr,g_qs,g_qg,g_qnr,g_w,mype)
+  !$$$  subprogram documentation block
+  !                .      .    .                                       .
+  ! subprogram:    general_read_wrf_mass_dirZDA  read arw model ensemble members
+  !   prgmmr: mizzi            org: ncar/mmm            date: 2010-08-11
+  !
+  ! abstract: read ensemble members from the arw model in "wrfout" netcdf format
+  !           for use with hybrid ensemble option. 
+  !
+  ! program history log:
+  !   2010-08-11  parrish, initial documentation
+  !   2010-09-10  parrish, modify so ensemble variables are read in the same way as in
+  !               subroutines convert_netcdf_mass and read_wrf_mass_binary_guess.
+  !               There were substantial differences due to different opinion about what
+  !               to use for surface pressure.  This issue should be resolved by coordinating
+  !               with Ming Hu (ming.hu@noaa.gov).  At the moment, these changes result in
+  !               agreement to single precision between this input method and the guess input
+  !               procedure when the same file is read by both methods.
+  !   2012-03-12  whitaker:  read data on root, distribute with scatterv.
+  !                          remove call to general_reload.
+  !                          simplify, fix memory leaks, reduce memory footprint.
+  !                          use genqsat, remove genqsat2_regional.
+  !                          replace bare 'stop' statements with call stop2(999).
+  !   2017-03-23  Hu      - add code to use hybrid vertical coodinate in WRF MASS core
+  !   2019-04-22  CAPS(C. Tong)   - add direct radar DA option
+  !   input argument list:
+  !
+  !   output argument list:
+  !
+  ! attributes:
+  !   language: f90
+  !   machine:  ibm RS/6000 SP
+  !
+  !$$$ end documentation block
+  
+      use netcdf, only: nf90_nowrite
+      use netcdf, only: nf90_open,nf90_close
+      use netcdf, only: nf90_inq_dimid,nf90_inquire_dimension
+      use netcdf, only: nf90_inq_varid,nf90_inquire_variable,nf90_get_var
+      use kinds, only: r_kind,r_single,i_kind
+      use gridmod, only: nsig
+      use constants, only: zero,one,fv,zero_single
```

Comment:
The variable, fv, has been identified as unused.  Please remove unused variable.

Reply:
We removed that variable as you suggested.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-f4844e36d71565749a9ceb0a3aa45ace24257a80b593048fbc2b5585e0bbca8fR1784